### PR TITLE
feat: initial based avatars (web&mobile)

### PR DESCRIPTION
**Pull request by zbettenbuk**

### CODE REVIEW

File: react/features/base/avatar/components/AbstractAvatar.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+            // URLS changed, so we need to try to fetch them again.
+            // Eslint doesn't like this statement, but based on the React doc, it's safe if it's
+            // wrapped in a condition: https://reactjs.org/docs/react-component.html#componentdidupdate
+
+            // eslint-disable-next-line react/no-did-update-set-state
+            this.setState({
+                avatarFailed: false,
+                gravatarFailed: false
+            });
```
Comment (virtuacoplenny) [inline react/features/base/avatar/components/AbstractAvatar.js]: Here's an alternative that came to mind and I haven't thought through but want to share: what if there were an image component that took in an array of urls and a placeholder component, and that component knew how to fallback and that component only updated when its props changed. That would eliminate this two-boolean handling.

File: react/features/base/avatar/components/web/Avatar.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    /**
+     * Implements {@code AbstractAvatar#_renderGravatar}.
+     *
+     * @inheritdoc
+     */
+    _renderInitialsAvatar(initials, color) {
+        return (
+            <div
+                className = { this._getAvatarClassName() }
```
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/base/avatar/components/web/Avatar.js]: support@jitsi.org does have a gravatar, FWIW.

File: modules/UI/UI.js [hunk 3]
```diff
@@ -510,8 +510,8 @@ UI.dockToolbar = dock => APP.store.dispatch(dockToolbox(dock));
  * @param {string} avatarURL - The URL to avatar image to display.
  * @returns {void}
  */
-UI.refreshAvatarDisplay = function(id, avatarURL) {
-    VideoLayout.changeUserAvatar(id, avatarURL);
+UI.refreshAvatarDisplay = function(id) {
```
[the remark this answers is not in the document]
Comment (saghul) [reply modules/UI/UI.js]: I agree in general. If you can reduce code from modules/ better!

File: modules/UI/videolayout/SmallVideo.js [hunk 4]
```diff
@@ -609,19 +607,21 @@ SmallVideo.prototype.updateView = function() {
...
     if (thumbnail) {
         ReactDOM.render(
-            <AvatarDisplay
-                className = 'userAvatar'
-                uri = { avatarUrl } />,
+            <Provider store = { APP.store }>
+                <AvatarDisplay
+                    className = 'userAvatar'
+                    participantId = { this.id }
+                    size = { 34 } />
```
[the remark this answers is not in the document]
Comment (saghul) [reply modules/UI/videolayout/SmallVideo.js]: Aren't the thumbnails the same size? I do agree that they should auto-size, otherwise tile view feels a bit weird with small avatars.

File: react/features/base/avatar/components/AbstractAvatar.js [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+    /**
+     * The size of the avatar.
+     */
+    size: number,
+
+    /**
+     * URI of the avatar, if any.
+     */
+    uri: ?string,
```
Comment (saghul) [inline react/features/base/avatar/components/AbstractAvatar.js]: This and _avatarURL seem to mean the same thing. Can we perhaps clarify by using some other name or coalesce both? Also, URI vs URL, let's be consistent.

File: react/features/base/avatar/components/AbstractAvatar.js [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    /**
+     * The string we base the initials on (this is generated from a list of precendences).
+     */
+    _initialsBase: ?string,
+
+    /**
+     * A string to override the initials to generate a color of. This is handy if you don't want to make
+     * the background color match the string that the initials are generated from.
+     */
+    colorBase?: string,
```
Comment (saghul) [inline react/features/base/avatar/components/AbstractAvatar.js]: Is this for the replaceement of AvtarListItem, so rooms in the same server get the same color? I find this and the initialsBase props quite confusing (but it's late so it may be that I'm tired).

File: react/features/base/avatar/components/AbstractAvatar.js (hunk 1 above)
Comment (saghul) [reply react/features/base/avatar/components/AbstractAvatar.js]: That sounds like a good idea!

File: react/features/base/avatar/components/AbstractAvatar.js [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+        const {
+            avatarFailed,
+            gravatarFailed
+        } = this.state;
+
+        const _initials = getInitials(_initialsBase);
+
+        // Precendence logic of avatars:
+
+        if (_avatarURL && _avatarURL.match(/^http[s]*:\/\//i) && !avatarFailed) {
```
Comment (saghul) [inline react/features/base/avatar/components/AbstractAvatar.js]: This logic is reversed to how mobile avatars behave currently, which is IMHO the expectation: we first show the locally generaterd avatar, then, if / when the remote one is loaded we show that one instead.

Using the network conditioner to make your connection slow. Then we wouldn't show anything until the load attempt has succeeded or failed.

File: react/features/base/avatar/components/AbstractAvatar.js [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+ * Maps part of the Redux state to the props of this component.
+ *
+ * @param {Object} state - The Redux state.
+ * @param {Props} ownProps - The own props of the component.
+ * @returns {Props}
+ */
+export function _mapStateToProps(state: Object, ownProps: Props) {
+    const { participantId } = ownProps;
+    const _participant = participantId && getParticipantById(state, participantId);
+    const _initialsBase = (_participant && (_participant.name || _participant.email)) || ownProps.displayName;
```
Comment (saghul) [inline react/features/base/avatar/components/AbstractAvatar.js]: I think we shouldn't use the email as initials.

Comment (saghul) [reply react/features/base/avatar/components/AbstractAvatar.js]: Nit: you can deconstruct displayName at the top toghether with participantId.

File: react/features/base/avatar/components/AbstractAvatar.js [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+
+    return {
```
Comment (saghul) [inline react/features/base/avatar/components/AbstractAvatar.js]: We must consider `disableThirdPartyRequests` here.

File: react/features/base/avatar/components/native/Avatar.js [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+     */
+    style?: StyleType
+}
+
+export const DEFAULT_AVATAR = require('../../../../../../images/avatar.png');
+
+/**
+ * Implements an avatar component that has 4 ways to render an avatar:
+ *
+ * - Based on an explicit avatar URI, if provided
```
Comment (saghul) [inline react/features/base/avatar/components/native/Avatar.js]: when are we using this first one, instead of using a participant?

File: react/features/base/avatar/components/AbstractAvatar.js [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+
+        // Precendence logic of avatars:
+
+        if (_avatarURL && _avatarURL.match(/^http[s]*:\/\//i) && !avatarFailed) {
+            // URL-based avatar first, if set and succeeds to load
+            return this._renderURLAvatar(_avatarURL);
+        } else if (_gravatarURL && !gravatarFailed) {
+            // Then gravatar if there is no URL based avatar, and there is email address
+            // and gravatar succeeds to load
+            return this._renderGravatar(_gravatarURL);
```
Comment (saghul) [inline react/features/base/avatar/components/AbstractAvatar.js]: why are we passing `_gravatarURL` to the function? it could read it from `props` like other parameters such as the size.

File: react/features/base/avatar/components/AbstractAvatar.js (hunk 1 above)
Comment (saghul) [reply react/features/base/avatar/components/AbstractAvatar.js]: I like the fact that supporting Gravatar (or any other avatar service) would just be a matter of adding one more URL to the list, making the whole thing more flexible.

File: react/features/base/avatar/components/native/Avatar.js [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+    _onGravatarLoadError: () => void;
+
+    /**
+     * Implements {@code AbstractAvatar#_renderDefaultAvatar}.
+     *
+     * @inheritdoc
+     */
+    _renderDefaultAvatar() {
+        return this._wrapAvatar(
+            <Image
```
Comment (saghul) [inline react/features/base/avatar/components/native/Avatar.js]: Why not FastImage?

File: react/features/base/avatar/components/native/RemoteAvatar.js [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+     * URI of the avatar to load.
+     */
+    uri: ?string
+};
+
+/**
+ * Implements a private class that is used to fetch and render remote avatars based on an URI.
+ */
+export default class RemoteAvatar extends PureComponent<Props> {
```
Comment (saghul) [inline react/features/base/avatar/components/native/RemoteAvatar.js]: random: we should really re-evaluate using functional components...

File: react/features/base/avatar/components/native/RemoteAvatar.js [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+
+    /**
+     * Size of the avatar.
+     */
+    size: number,
+
+    /**
+     * URI of the avatar to load.
+     */
+    uri: ?string
```
Comment (saghul) [inline react/features/base/avatar/components/native/RemoteAvatar.js]: This cannot be optional, can it?

File: react/features/base/avatar/components/web/Avatar.js [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+class Avatar extends AbstractAvatar<Props> {
+
+    /**
+     * Constructs a style object to be used on the avatars.
+     *
+     * @param {string?} color - The desired background color.
+     * @returns {Object}
+     */
+    _getAvatarStyle(color) {
+        const { size = DEFAULT_SIZE } = this.props;
```
Comment (saghul) [inline react/features/base/avatar/components/web/Avatar.js]: Use `defaultProps` instead.

File: react/features/base/avatar/components/web/Avatar.js [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+
+        return {
+            backgroundColor: color || undefined,
```
Comment (saghul) [inline react/features/base/avatar/components/web/Avatar.js]: Why is the background color optional?

File: react/features/base/avatar/functions.js [hunk 17]
```diff
@@ -0,0 +1,54 @@
+// @flow
+
+import _ from 'lodash';
+
+const AVATAR_COLORS = [
```
Comment (saghul) [inline react/features/base/avatar/functions.js]: Can we have some more? 5 seeems to low.

File: react/features/base/participants/components/ParticipantView.native.js [hunk 18]
```diff
@@ -265,45 +249,14 @@ class ParticipantView extends Component<Props> {
...
-        // But the local participant changes her ID as she joins, leaves.
-        // TODO @lyubomir: The participants may change their avatar URLs at
-        // runtime which means that, if their old and new avatar URLs fail to
-        // download, Avatar will change their automatically-generated colors.
-        avatar || participant.local || (avatar = `#${participant.id}`);
-
-        // ParticipantView knows before Avatar that an avatar URL will be used
-        // so it's advisable to prefetch here.
-        avatar && !avatar.startsWith('#')
-            && FastImage.preload([ { uri: avatar } ]);
```
Comment (saghul) [inline react/features/base/participants/components/ParticipantView.native.js]: This preload makes a difference. We should incorporaste it in the new implementation.

File: react/features/settings/components/web/ProfileTab.js [hunk 19]
```diff
@@ -75,8 +75,8 @@ class ProfileTab extends AbstractDialogTab<Props> {
     render() {
         const {
             authEnabled,
-            displayName,
-            email,
+            displayName = '',
```
Comment (saghul) [inline react/features/settings/components/web/ProfileTab.js]: why is this needed? Maybe use `defaultProps` if need be.

File: react/features/toolbox/components/web/OverflowMenuProfileItem.js [hunk 20]
```diff
@@ -80,7 +75,9 @@ class OverflowMenuProfileItem extends Component<Props> {
                 className = { classNames }
                 onClick = { this._onClick }>
                 <span className = 'overflow-menu-item-icon'>
-                    <Avatar uri = { avatarURL } />
+                    <Avatar
+                        participantId = { _localParticipant.id }
+                        size = { 24 } />
```
Comment (saghul) [inline react/features/toolbox/components/web/OverflowMenuProfileItem.js]: Extract constant to styles

File: react/features/welcome/components/WelcomePageSideBar.native.js [hunk 21]
```diff
@@ -43,14 +42,14 @@ type Props = {
...
      */
-    _avatarURL: string,
+    _displayName: string,
 
     /**
-     * Display name of the local participant.
+     * The local participant.
      */
-    _displayName: string,
+    _localParticipant: Object,
```
Comment (saghul) [inline react/features/welcome/components/WelcomePageSideBar.native.js]: Why are we passing the entire participant instead of just the ID?

File: modules/UI/videolayout/SmallVideo.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (virtuacoplenny) [reply modules/UI/videolayout/SmallVideo.js]: Thumbnails are not the same size. In vertical filmstrip mode, remote thumbnails are actually taller than the local thumbnail. (There is also a bug where thumbnail size is adjustable in interface config but that was never well supported in vertical filmstrip, and no one has complained to me so far.) Thumbnails in horizontal filmstrip mode, the one that was advertised for use with the iframe api, resize to fit the width of the window. And like you said tile view avatars resize to fit the width of the window.

File: modules/UI/UI.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (virtuacoplenny) [reply modules/UI/UI.js]: So it turns out this isn't as easy to kill off right now, I think. Maybe with additional changes it might be, but looking at the branch in its current state, the reason this call is needed is really to tell the local video to re-render its avatar because the local participant got a new id (joined the conference). Without this method, the local video will continue to use some initial default id, leaving the avatar to also use the initial default id instead of the local participant id stored in redux.

File: css/main.scss [hunk 22]
```diff
@@ -86,5 +86,6 @@ $flagsImagePath: "../images/";
 @import 'navigate_section_list';
 @import 'third-party-branding/google';
 @import 'third-party-branding/microsoft';
+@import 'avatar';
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply css/main.scss]: Fixed, thanks!

File: react/features/base/avatar/components/AbstractAvatar.js [hunk 23]
```diff
@@ -0,0 +1,215 @@
...
+
+export type Props = {
+
+    /**
+     * URI of the avatar, if identifiable.
+     */
+    _avatarURL: ?string,
+
+    /**
+     * URI of the gravatar if identifyable.
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/base/avatar/components/AbstractAvatar.js]: Removed, thanks

File: modules/UI/videolayout/SmallVideo.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply modules/UI/videolayout/SmallVideo.js]: Added support for 100%, thanks for reporting.

Though, the font sizing could be improved, but looks okay for now

File: react/features/base/avatar/components/AbstractAvatar.js (hunk 1 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/AbstractAvatar.js]: This went to a new direction, so kinda obsolete now. We now have the array of urls as described above, but there is a prefetch function that does all these and stores the result in Redux, so then we can even signal the avatar change through the external API (this couldn't have been done with a UI element)

File: react/features/base/avatar/components/AbstractAvatar.js (hunk 5 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/AbstractAvatar.js]: This got fully refactored, thanks.

Though I'll keep an eye on url vs uri. URI is my pick though as it can also identify other resources as well

File: react/features/base/avatar/components/AbstractAvatar.js (hunk 6 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/AbstractAvatar.js]: Yes, that is for the avatar list item, where we want to color things differently. The xxxBase postfix I use to denote something that some other thing will be "based on". E.g. initialsBase is something we base the initials on (e.g. this can be the name or the email). colorBase is the basis of the color, if provided

File: react/features/base/avatar/functions.js (hunk 17 above)
Comment (zbettenbuk) [reply react/features/base/avatar/functions.js]: I agree. We can have as many as we want, but we're waiting for Tibi to provide final colors. Yana said that we can go with these for the time being. There will be a folow up "color clean up" PR soon that will change other colors as well

File: react/features/base/participants/components/ParticipantView.native.js (hunk 18 above)
Comment (zbettenbuk) [reply react/features/base/participants/components/ParticipantView.native.js]: Yes, I wanted to incorporate that, but as you can see the preload went to another direction and FastImage doesn't have a similar promise based prefetch like the standard Image has. I'm open for any suggestion here though

File: react/features/settings/components/web/ProfileTab.js (hunk 19 above)
Comment (zbettenbuk) [reply react/features/settings/components/web/ProfileTab.js]: This was needed to fix a bug on the avatar initials update on editing the displayName (remove it and you'll see the warning generated by Redux). I'll add them to defaultProps though

File: react/features/toolbox/components/web/OverflowMenuProfileItem.js (hunk 20 above)
Comment (zbettenbuk) [reply react/features/toolbox/components/web/OverflowMenuProfileItem.js]: But that's web. I don't want to start using styles.js on web. Or did you mean something else?

File: react/features/welcome/components/WelcomePageSideBar.native.js (hunk 21 above)
Comment (zbettenbuk) [reply react/features/welcome/components/WelcomePageSideBar.native.js]: Dunno, because it was available. But you are right, I change it, thanks!

File: react/features/base/avatar/components/AbstractAvatar.js (hunk 7 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/AbstractAvatar.js]: Now this was refactored, thanks.

There is a caveat though: if there is an URI provided to the avatar component, we still try to load that first, before rendering any default (or initials based) avatar. In this case we don't render anything until the uri deems to be loaded or fails. The reason for this is that this is a corner case (e.g. in case of IncomingCallPage) and since the component is the one that decides if the URI can be loaded or not, we need to render the avatar (empty at first) and let them succeed or fail. Then if we render an initials based avatar, there is a short period (blink) when the avatar already loaded and rendered and the initials based avatar is still shown, so users see a duplicated avatar. I think that is a worse UX than not rendering anything for a while. But again, this is a really rare corner case.

File: react/features/base/avatar/components/AbstractAvatar.js (hunk 9 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/AbstractAvatar.js]: Added it to the preloader on web

File: react/features/base/avatar/components/AbstractAvatar.js (hunk 8 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/AbstractAvatar.js]: I added the deconstruct, thanks.

Why do you think we shouldn't use the email for initials?

File: react/features/base/avatar/components/native/Avatar.js (hunk 10 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/native/Avatar.js]: E.g. in case of the IncomingCallPage

File: react/features/base/avatar/components/AbstractAvatar.js (hunk 11 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/AbstractAvatar.js]: That code is now totally different.

There is still a similar call in there though. I did that because I felt it more consistent. We want to render an avatar with some details provided that may or may not be available at the place of render. In this case it is, true, but in case of the other call, it's not. So I went for function footprint consistency instead of minimalistic. I guess it's a matter of taste

File: react/features/base/avatar/components/native/Avatar.js (hunk 12 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/native/Avatar.js]: I answered it in another thread. In short, we now prefetch the urls with this component, so it would be a duplication do render it with another

File: react/features/base/avatar/components/native/RemoteAvatar.js (hunk 13 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/native/RemoteAvatar.js]: I agree

File: react/features/base/avatar/components/native/RemoteAvatar.js (hunk 14 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/native/RemoteAvatar.js]: Nope, thanks

File: react/features/base/avatar/components/web/Avatar.js (hunk 15 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/web/Avatar.js]: That is not there anymore :)

File: react/features/base/avatar/components/web/Avatar.js (hunk 16 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/web/Avatar.js]: Because then we use the default one that is defined in the class

File: react/features/base/participants/middleware.js [hunk 24]
```diff
@@ -341,19 +340,14 @@ function _participantJoinedOrUpdated({ getState }, next, action) {
...
 
-        if (oldAvatarURL !== newAvatarURL) {
-            const currentKnownId = local ? APP.conference.getMyUserId() : id;
+        const currentKnownId = local ? APP.conference.getMyUserId() : id;
 
-            APP.UI.refreshAvatarDisplay(currentKnownId, newAvatarURL);
-            APP.API.notifyAvatarChanged(currentKnownId, newAvatarURL);
-        }
+        APP.UI.refreshAvatarDisplay(currentKnownId);
+        APP.API.notifyAvatarChanged(currentKnownId);
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/base/participants/middleware.js]: I guess this is now outdated after your recent change, isn't it?

File: react/features/base/avatar/components/AbstractAvatar.js [hunk 25]
```diff
@@ -0,0 +1,215 @@
...
+    /**
+     * Function to render the actual, plarform specific initials-based avatar component.
+     *
+     * @param {string} initials - The initials to use.
+     * @param {string} color - The color to use.
+     * @returns {React$Element<*>}
+     */
+    _renderInitialsAvatar: (string, string) => React$Element<*>
+
+    /**
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/base/avatar/components/AbstractAvatar.js]: that's how we spell it here, okay?

**Comment by zbettenbuk** [comment]
Comments addressed, tests fixed

File: modules/UI/videolayout/SmallVideo.js [hunk 26]
```diff
@@ -609,19 +607,22 @@ SmallVideo.prototype.updateView = function() {
  * Updates the react component displaying the avatar with the passed in avatar
  * url.
  *
- * @param {string} avatarUrl - The uri to the avatar image.
  * @returns {void}
  */
-SmallVideo.prototype.avatarChanged = function(avatarUrl) {
+SmallVideo.prototype.avatarChanged = function() {
```
Comment (virtuacoplenny) [inline modules/UI/videolayout/SmallVideo.js]: Should this be changed to something like initializeAvatar?

File: react/features/base/avatar/components/AbstractAvatar.js [hunk 27]
```diff
@@ -0,0 +1,177 @@
...
+        if (prevProps.uri !== this.props.uri) {
+
+            // URI changed, so we need to try to fetch it again.
+            // Eslint doesn't like this statement, but based on the React doc, it's safe if it's
+            // wrapped in a condition: https://reactjs.org/docs/react-component.html#componentdidupdate
+
+            // eslint-disable-next-line react/no-did-update-set-state
+            this.setState({
+                avatarFailed: false
+            });
```
Comment (virtuacoplenny) [inline react/features/base/avatar/components/AbstractAvatar.js]: What if it it started as failed and a callback was called on both success and failed to set the state, so something like avatarLoaded instead of avatarFailed? That would remove setting state in _onAvatarLoadError.

File: react/features/base/avatar/components/AbstractAvatar.js [hunk 28]
```diff
@@ -0,0 +1,177 @@
...
+
+    /**
+     * Implements {@code Componenr#render}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const {
+            _initialsBase,
+            _loadableUrl,
```
Comment (virtuacoplenny) [inline react/features/base/avatar/components/AbstractAvatar.js]: You sure you want to keep the name loadableUrl? Why double up on uri and loadableUrl?

File: react/features/base/participants/middleware.js [hunk 29]
```diff
@@ -330,26 +332,28 @@ function _participantJoinedOrUpdated({ getState }, next, action) {
...
-        if (oldAvatarURL !== newAvatarURL) {
-            const currentKnownId = local ? APP.conference.getMyUserId() : id;
+        getFirstLoadableAvatarUrl(updatedParticipant)
+            .then(url => {
+                dispatch(setLoadableAvatarUrl(id, url));
+            });
+    }
 
-            APP.UI.refreshAvatarDisplay(currentKnownId, newAvatarURL);
-            APP.API.notifyAvatarChanged(currentKnownId, newAvatarURL);
```
Comment (virtuacoplenny) [inline react/features/base/participants/middleware.js]: This call to APP.API still needs to be brought back in.

File: modules/UI/videolayout/SmallVideo.js (hunk 26 above)
Comment (zbettenbuk) [reply modules/UI/videolayout/SmallVideo.js]: Yep, I renamed it, thanks!

File: react/features/base/participants/middleware.js (hunk 29 above)
Comment (zbettenbuk) [reply react/features/base/participants/middleware.js]: Done by you, and thanks for it again!

File: react/features/base/avatar/components/AbstractAvatar.js (hunk 28 above)
Comment (zbettenbuk) [reply react/features/base/avatar/components/AbstractAvatar.js]: Renamed to _loadableAvatarUrl, thanks
